uwb: disable command/event filtering for D-Link DUB-1210

From: Anderson Lizardo <anderson.lizardo@indt.org.br>

The D-Link DUB-1210 HWA uses commands and events from the WHCI specification,
although reporting itself as WUSB compliant. Therefore, we disable WUSB
command/event filtering for it.

USB_DEVICE_AND_INTERFACE_INFO is used for matching only the RC interface.

Signed-off-by: Anderson Lizardo <anderson.lizardo@indt.org.br>
Signed-off-by: David Vrabel <david.vrabel@csr.com>
---
 drivers/uwb/hwa-rc.c |   13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

--- a/drivers/uwb/hwa-rc.c
+++ b/drivers/uwb/hwa-rc.c
@@ -62,6 +62,9 @@
 #define D_LOCAL 1
 #include <linux/uwb/debug.h>
 
+/* The device uses commands and events from the WHCI specification, although
+ * reporting itself as WUSB compliant. */
+#define WUSB_QUIRK_WHCI_CMD_EVT		0x01
 
 /**
  * Descriptor for an instance of the UWB Radio Control Driver that
@@ -824,7 +827,11 @@ static int hwarc_probe(struct usb_interf
 		goto error_neep_init;
 	}
 	hwarc->uwb_rc = uwb_rc;
-	result = uwb_rc_add(uwb_rc, dev, hwarc, hwarc_cmd,
+	if (id->driver_info & WUSB_QUIRK_WHCI_CMD_EVT)
+		result = uwb_rc_add(uwb_rc, dev, hwarc, hwarc_cmd,
+			NULL, NULL);
+	else
+		result = uwb_rc_add(uwb_rc, dev, hwarc, hwarc_cmd,
 			hwarc_filter_cmd, hwarc_filter_event);
 	if (result < 0)
 		goto error_rc_add;
@@ -866,6 +873,10 @@ static void hwarc_disconnect(struct usb_
 
 /** USB device ID's that we handle */
 static struct usb_device_id hwarc_id_table[] = {
+	/* D-Link DUB-1210 */
+	{ USB_DEVICE_AND_INTERFACE_INFO(0x07d1, 0x3d02, 0xe0, 0x01, 0x02),
+	  .driver_info = WUSB_QUIRK_WHCI_CMD_EVT },
+	/* Generic match for the Radio Control interface */
 	{ USB_INTERFACE_INFO(0xe0, 0x01, 0x02), },
 	{ },
 };
